modules/RTC/ScreenObtainer.js

```diff
@@ -423,6 +418,37 @@ const ScreenObtainer = {
                 this.checkForChromeExtensionOnInterval(options,
                     streamCallback, failCallback);
             });
+    },
+
+    /**
+     * Obtains a screen capture stream using getDisplayMedia.
+     *
+     * @param callback - The success callback.
+     * @param errorCallback - The error callback.
+     */
+    obtainScreenFromGetDisplayMedia(options, callback, errorCallback) {
+        navigator.getDisplayMedia({ video: true })
+            .then(stream => {
+                let applyConstraintsPromise;
+
+                if (stream
+                    && stream.getTracks()
+                    && stream.getTracks().length > 0) {
+                    applyConstraintsPromise = stream.getTracks()[0]
+                        .applyConstraints(options.trackOptions);
```

Comment:
i don't think i saw why we have to do this after (apply the constraints) as opposed to setting these in the getDisplayMedia call, does the API not work that way?

Reply:
No the, spec constraints that you can pass to getDisplayMedia are audio: true/false, video:true: false and some os specific about surface and cursor.
I was checking the issue tracker of the spec about the framerate and this is what they say should be done and also mentioned here: https://webrtchacks.com/chrome-screensharing-getdisplaymedia/
https://github.com/w3c/mediacapture-screen-share/issues/49